mlserver/registry.py

```diff
@@ -225,20 +196,15 @@ async def get_model(self, version: str = None) -> MLModel:
 
     async def get_models(self) -> List[MLModel]:
         # NOTE: `.values()` returns a "view" instead of a list
-        models = list(self._versions.values())
-
-        # Add default if not versioned (as it won't be present on the
-        # `_versions` dict
-        if self.default and not self.default.version:
-            models.append(self.default)
-
-        return models
+        return list(self._versions.values())
```

Comment:
nit: would this return the default model twice if it also has a version?

Reply:
I believe this isn't possible since values are stored in `dict`.

Reply:
Wouldn't they get saved under different keys though? As in,

```python
self._versions = {
	"v1.1.1": model1,
	None: model1
}

self._versions.values() # then returns -> [model1, model1]
```

Reply:
What about this case?

```python
self._versions = {
	"v1.1.1": model1,
        "v2.1.1": model1,
	None: model1
}

self._versions.values() # then returns -> [model1, model1, model1]```
```
I don't understand the use-case, so I don't have any alternative.

Reply:
Well, a model object can only have one version, so `model1` couldn't have two different versions. However, a model can have a version **and** be the "default" model at the same time.

So I guess the question is, following your changes, is the previous case possible? (i.e. having something like `{"v1": model1, None: model1}`)

Reply:
Thanks for explaining.
I tried to preserve previous behavior, In previous version of code if there were a default it append the default to the end of `models` and return it.
Unfortunately I didn't understand the code enough to see any problem in this case.

Reply:
Yeah, totally understand that. The previous code was a bit convoluted, and wasn't easy to understand. The new version is way cleaner.

Regarding the potential duplicated model, that was essentially the main reason of this `self.default.version` check (i.e. if the default model also has a version, don't include it, as it will already be present on the `self._versions` dictionary). 

https://github.com/SeldonIO/MLServer/pull/757/files#diff-8d68e1b9bf57de7aa43eaed4717d3fc4c14241afef15576891e1ad27d74322a5L230-L233

Perhaps what would be best is to add a test case covering that it doesn't happen after the changes? (i.e. that `registry.get_models()` doesn't return the same model twice)